framework/src/relationshipmanagers/ElementSideNeighborLayers.C

```diff
@@ -74,7 +74,7 @@ ElementSideNeighborLayers::internalInit()
   if (executioner_ptr)
   {
     auto & fe_problem = executioner_ptr->feProblem();
-    auto & nl_sys = fe_problem.getNonlinearSystem();
+    auto & nl_sys = fe_problem.getNonlinearSystemBase();
```

Comment:
The code was so bad. You even directly call getNonlienarSystem.

Reply:
> getNonlienarSystem

No he didn't 😄

Reply:
This was not my code:
```c++
lindad@Alexanders-Mac-Pro:~/projects/moose/framework/src/relationshipmanagers(next)$ git blame -L:internalInit ElementSideNeighborLayers.C
556a9d8300b (Derek Gaston 2018-10-19 16:30:33 -0600 66) ElementSideNeighborLayers::internalInit()
a099f90c395 (Cody Permann 2018-01-04 11:12:03 -0700 67) {
886fc07a519 (Derek Gaston 2018-10-22 18:46:45 -0600 68)   auto functor = libmesh_make_unique<DefaultCoupling>();
886fc07a519 (Derek Gaston 2018-10-22 18:46:45 -0600 69)   functor->set_n_levels(_layers);
886fc07a519 (Derek Gaston 2018-10-22 18:46:45 -0600 70) 
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 71)   // Need to see if there are periodic BCs - if so we need to dig them out
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 72)   auto executioner_ptr = _app.getExecutioner();
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 73) 
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 74)   if (executioner_ptr)
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 75)   {
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 76)     auto & fe_problem = executioner_ptr->feProblem();
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 77)     auto & nl_sys = fe_problem.getNonlinearSystem();
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 78)     auto & dof_map = nl_sys.dofMap();
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 79)     auto periodic_boundaries_ptr = dof_map.get_periodic_boundaries();
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 80) 
8917821f9d7 (Cody Permann 2019-04-16 11:28:23 -0600 81)     mooseAssert(periodic_boundaries_ptr, "Periodic Boundaries Pointer is nullptr");
8917821f9d7 (Cody Permann 2019-04-16 11:28:23 -0600 82) 
8917821f9d7 (Cody Permann 2019-04-16 11:28:23 -0600 83)     functor->set_mesh(&_mesh.getMesh());
8917821f9d7 (Cody Permann 2019-04-16 11:28:23 -0600 84)     functor->set_periodic_boundaries(periodic_boundaries_ptr);
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 85)   }
bf3ca159732 (Derek Gaston 2018-11-19 08:32:23 -0700 86) 
886fc07a519 (Derek Gaston 2018-10-22 18:46:45 -0600 87)   _functor = std::move(functor);
a099f90c395 (Cody Permann 2018-01-04 11:12:03 -0700 88) }
a0b1c7a1a88 (Alex Lindsay 2019-07-16 11:27:12 -0700 89) 
```
It just happened that with my changes this code path now got hit for your Eigen stuff. So you can calm down sir! 😄